### PR TITLE
chore(sdk): Fix bad test that reads from stdin

### PR DESCRIPTION
**Pull request by timoffex**
Description
-----------

Explicitly replace `stdin` by `None` as a way to simulate there not being a TTY available. `pytest` is supposed to replace `stdin`, but there's no guarantee that it replaces it with something that doesn't _look_ like a TTY. In fact, when I try to run this test locally, it prompts me for input!

See the second failure here: https://app.circleci.com/pipelines/github/wandb/wandb/31659/workflows/f709e516-f2f4-4ae0-ad51-836af1b880b4/jobs/1052367/tests. The `login()` routine tries to prompt using `input()` which raises an EOF error. I think the code might not be testing for a TTY correctly, but it's hard to follow.

- [x] I updated CHANGELOG.md, or it's not applicable

### CODE REVIEW

File: tests/pytest_tests/unit_tests/test_public_api.py [hunk 1]
```diff
@@ -7,7 +7,8 @@
 from wandb.sdk.internal.thread_local_settings import _thread_local_api_settings
 
 
-def test_api_auto_login_no_tty():
+def test_api_auto_login_no_tty(monkeypatch):
+    monkeypatch.setattr("sys.stdin", None)
```
Comment (kptkin) [inline tests/pytest_tests/unit_tests/test_public_api.py:11]: would you mind using a contextmanager  and `unittest.mock` instead?

**Review by kptkin — APPROVED** [review]

File: tests/pytest_tests/unit_tests/test_public_api.py (hunk 1 above)
Comment (timoffex) [reply tests/pytest_tests/unit_tests/test_public_api.py]: done!
